templates/cdap-deployment.yaml

```diff
@@ -110,6 +117,12 @@ spec:
             - name: cdap-se-vol-{{$k}}
               mountPath: {{$v}}
             {{end}}
+          {{if $.Base.SecurityContext}}
+          securityContext:
+            privileged: {{$.Base.SecurityContext.Privileged}}
```

Comment:
If some fields of security are left unspecified, I think we will use default value for them. Just wanted to make sure we thought about it. Default values look fine to me. Please double check

Reply:
Good point, we need to be careful of the default behaviors. I've gone ahead and modified it so that:
 - If the integer values (runAsUser, runAsGroup, and fsGroup) are not specified (or zero) in the cdap-cr, no value is specified in the resulting yaml.
 - If the boolean values (allowPrivilegeEscalation, privileged, runAsNonRoot, readOnlyFilesystem) are not specified in the cdap-cr, they are explicitly set to default values defined by the Kubernetes API, i.e. allowPrivilegeEscalation = true, privileged = false, runAsNonRoot = false, readOnlyFilesystem = false.